apps/dashboard/app/models/script.rb

```diff
@@ -45,12 +45,15 @@ def next_id(project_dir)
     end
   end
 
+  attr_reader :project_dir, :smart_attributes
+
   def initialize(opts = {})
     opts = opts.to_h.with_indifferent_access
 
     @project_dir = opts[:project_dir] || raise(StandardError, 'You must set the project directory')
     @id = opts[:id]
     @title = opts[:title].to_s
+    @smart_attributes = opts[:form] || []
```

Comment:
You'll still have to actually build the smart attributes here. `@smart_attributes` is an array of `SmartAttributes`.

That is, we have to do something along the lines of what `batch_connect` apps do here.  That is, given the form structure from the YAML they generate an array of `SmartAttributes`.  

(note we're not accounting for the `attribute` structure just yet, only `form`).
https://github.com/OSC/ondemand/blob/d66ce3cb6999c42dfbfaa028981c2b08658090ca/apps/dashboard/app/models/batch_connect/app.rb#L212-L232

Reply:
What is a SmartAttribute object in the code? I can't find it.

Reply:
`SmartAttributes` is the module in `app/lib/smart_attributes.rb`. Looks like they all inherit from `SmartAttributes::Attribute``. 

https://github.com/OSC/ondemand/blob/d66ce3cb6999c42dfbfaa028981c2b08658090ca/apps/dashboard/app/lib/smart_attributes/attributes/auto_accounts.rb#L31

This is the important line. This is how you actually build one of these objects, through this factory.  It'll figure out what object to actually build given an id and options. `form` is a list of ids. `attributes` is a hash of options. But again, we're not using/focusing on `attributes` just yet.
```
SmartAttributes::AttributeFactory.build(attribute_id, attribute_opts) 
```

Reply:
Is it a convention or something? I don't know what the phrase means.

Reply:
> I don't know what the phrase means.

What phrase?  I mispoke when I said `SmartAttributes` and should have said `SmartAttributes::Attribute`.

so @smart_attribute is an array of `SmartAttributes::Attribute` objects (or their child classes like `SmartAttribute::AutoAccounts` above).

Reply:
so should the `:smart_attributes` reader be removed, then replaced with a method like `smart_attributes` that returns an array built after iterating over the `form` key/value pairs, but only selecting what is found under `form` key? Is form overloaded? Does it mean both a web form and key to use for the `smart_attributes` values?

Reply:
so should I add a method to this `SmartAttribute` module to get an array for `@smart_attributes`? I don't see anything in that code for building an array of these. I don't understand the idea of this so far.

Reply:
Well, i guess it isn't a `SmartAttribute` object, just `attributes` but I shouldn't focus on that, so now im confused.

Reply:
Look at how `batch_connect` apps do this in the attribute function I gave above.

I believe it's something like this, just mapping the array of form items to an array of `SmartAttributes::Attribute` through the `SmartAttributes::AttributeFactory`.

```ruby
def initialize(opts)
  @smart_attributes = build_smart_attributes(opts[:form] || [])
end

def build_smart_attributes(form_list)
  form_list.map do |form_item_id|
     SmartAttributes::AttributeFactory.build(form_item_id, {}) 
  end
end
```

Batch connect apps do this now.  This is what we're taking inspiration from. Will we use the same exact code? No, but it's essentially the same functionality. 

* Given a YAML that defines `form` as a list of ids
* generate an array of `SmartAttributes::Attribute`s in the model based off of those ids
* Show that array pf `SmartAttributes::Attribute`s in the view as an HTML form

This is more or less how batch connect apps work to create a web form from the YAML file. We're trying to replicate the same behavior.